### PR TITLE
[WFLY-3990] Add better documentation to indicate what the values for …

### PR DESCRIPTION
**Pull request by jamezp**
…the attributes actually do for queuing tasks.

Replaces https://github.com/wildfly/wildfly/pull/8167.

Improved documentation for the value of the `queue-length` and `core-threads` attributes. 

Changed the `core-threads` attribute to allow `null` (`UNDEFINED`) values. If found to be `null` the value is calculated based on the number of threads. Due to this change the default value was removed from the standard configuration. Assuming the defaults are used the thread-pool will be unbounded by default.

### CODE REVIEW

File: ee/src/main/java/org/jboss/as/ee/subsystem/ManagedExecutorServiceAdd.java [hunk 1]
```diff
@@ -58,12 +59,35 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
...
         final TimeUnit keepAliveTimeUnit = TimeUnit.MILLISECONDS;
         final long threadLifeTime = 0L;
-        final int queueLength = ManagedExecutorServiceResourceDefinition.QUEUE_LENGTH_AD.resolveModelAttribute(context, model).asInt();
+
+        final int queueLength;
+        final ModelNode queueLengthModel = ManagedExecutorServiceResourceDefinition.QUEUE_LENGTH_AD.resolveModelAttribute(context, model);
+        if (queueLengthModel.isDefined()) {
+            queueLength = queueLengthModel.asInt();
+        } else {
+            queueLength = Integer.MAX_VALUE;
```
Comment (bstansberry) [inline ee/src/main/java/org/jboss/as/ee/subsystem/ManagedExecutorServiceAdd.java]: Why isn't this the default value on QUEUE_LENGTH_AD?

Comment (jamezp) [inline ee/src/main/java/org/jboss/as/ee/subsystem/ManagedExecutorServiceAdd.java]: @n1hility wanted to leave it `null`.

File: ee/src/main/resources/org/jboss/as/ee/subsystem/LocalDescriptions.properties [hunk 2]
```diff
@@ -35,10 +35,16 @@ managed-executor-service.context-service=The name of the context service to be u
...
+  default core-size is calculated based on the number of processors. A value of zero is not \
+  advised. See the queue-length attribute for details on how this value is used to determine the queuing strategy.
 managed-executor-service.max-threads=The maximum number of threads to be used by the executor.
 managed-executor-service.keepalive-time=When the number of threads is greater than the core, this is the maximum time, in milliseconds, that excess idle threads will wait for new tasks before terminating.
-managed-executor-service.queue-length=The executor's task queue capacity.
+managed-executor-service.queue-length=The executors task queue capacity. For an unbounded queue the 2147483647 (Integer.MAX_VALUE \
+  value must be used and a core-threads value greater than zero is required. A queue-length value of zero uses a direct \
+  hand-off strategy for task queue management. If the queue-length value is set to 2147483647 and the core-threads value is set \
+  to 0 a direct hand-off strategy is also used. Finally a bounded queue is used for all other valid settings. If the \
+  queue-length is left undefined 2147483647 is used.
```
Comment (n1hility) [inline ee/src/main/resources/org/jboss/as/ee/subsystem/LocalDescriptions.properties]: This should be updated to reflect the new contract we are trying to expose.

**Comment by bstansberry** [comment]
Looks good.

**Pull request merged**